packages/kbn-config/src/deprecation/apply_deprecations.test.ts

```diff
@@ -116,6 +116,36 @@ describe('applyDeprecations', () => {
     expect(migrated).toEqual({ foo: 'bar', newname: 'renamed' });
   });
 
+  it('works for nested properties', () => {
+    const initialConfig = {
+      foo: 'bar',
+      deprecated: { nested: 'deprecated' },
+      nested: {
+        from: {
+          rename: 'renamed',
+        },
+        to: {
+          keep: 'keep',
+        },
+      },
+    };
+
+    const { config: migrated } = applyDeprecations(initialConfig, [
+      wrapHandler(deprecations.unused('deprecated.nested')),
+      wrapHandler(deprecations.rename('nested.from.rename', 'nested.to.renamed')),
+    ]);
+
+    expect(migrated).toEqual({
```

Comment:
optional nit: https://jestjs.io/docs/expect#tostrictequalvalue works better here. `key: undefined` might be a critical problem for the config object.
The same for `packages/kbn-config/src/deprecation/unset_and_clean_empty_parent.test.ts` file

Reply:
good point! Copy-pasta error!